MatrixSDKTests/MXSpaceServiceTest.swift

```diff
@@ -169,4 +202,179 @@ class MXSpaceServiceTest: XCTestCase {
             }
         }
     }
+    
+    /// - Create Bob
+    /// - Setup Bob session
+    /// - Create root public space A
+    /// - Create a child space B with space A as parent
+    ///
+    /// -> Bob must see the child space state event of space B
+    func testAddChildSpace() throws {
+        self.doSpaceServiceTestWithBob(testCase: self) { (spaceService, session, expectation) in
+            
+            let expectedRootSpaceName = "Space A"
+            let expectedChildSpaceName = "Space B"
+            
+            self.createSpaces(with: spaceService, spaceNames: [
+                expectedRootSpaceName,
+                expectedChildSpaceName
+            ]) { (response) in
+                switch response {
+                case .success(let spaces):
+                    let rootSpace = spaces[0]
+                    let childSpace = spaces[1]
+                    
+                    let viaServers = [session.credentials.homeServerName()!]
+                    
+                    rootSpace.addChild(spaceId: childSpace.spaceId, viaServers: viaServers, order: nil, autoJoin: false, suggested: false) { (response) in
+                        switch response {
+                        case .success:
+                            
+                            // Make an initial sync and and get
+                            session.start { (response) in
+                                switch response {
+                                case .success:
+                                    
+                                    guard let foundRootSpace = spaceService.getSpace(withId: rootSpace.spaceId) else {
+                                        XCTFail("Fail to found the root space")
+                                        return
+                                    }
+                                    
+                                    foundRootSpace.room.state({ (roomState) in
+                                        
+                                        let stateEvent = roomState?.stateEvents(with: .spaceChild)?.first
+                                        
+                                        XCTAssert(stateEvent?.stateKey == childSpace.spaceId)
+                                        
+                                        expectation.fulfill()
+                                    })
+                                case .failure(let error):
+                                    XCTFail("Sync failed with error\(error)")
```

Comment:
Call `expectation.fulfill()` else the test will block the test suite until the expectation timeout.